### DISCUSSION TITLE
Sporadic error when publishing test results to Azure Pipelines

### DISCUSSION THREAD

**Discussion by GitSteller** [tcunit/TcUnit#197] [Q&A]
Hi together,
we are using the TcUnit-Runner in combination with an Azure pipeline. The single steps in the pipe are simple and similar to the example on the User manual of the TcUnit page:

<html>
<body>

<pre class="brush: yaml; title: ; notranslate" title="">
  | # Demo CI workflow with TcUnit-Runner
  | name: TcUnit-Runner
  |  
  | trigger:
  | - main
  | pr:
  | - main
  |  
  | pool: 'tcunit'    # self-hosted agent
  |  
  | steps:
  |  
  | - checkout: self
  | displayName: Checkout repository
  |  
  | - task: CmdLine@2
  | displayName: Run PLC unit-tests with TcUnit-Runner
  | condition: succeeded()
  | inputs:
  | failOnStderr: true
  | # Start TcUnit-Runner with a timeout of 15 min
  | script: '&quot;C:\Program Files (x86)\TcUnit-Runner\LaunchTcUnit.bat&quot; -U 15'
  |  
  | - task: PublishTestResults@2
  | displayName: Publish TcUnit results
  | condition: or(succeeded(),failed())
  | inputs:
  | testResultsFormat: 'JUnit'
  | testResultsFiles: '**/TcUnit_xUnit_results.xml'
  | </pre>

</body>
</html>

The most time when we run our pipeline, everything is working fine. But sometimes and very sporadic it throws us an error in the task "PublishTestResults@2". The error occurs at the beginning of the execution of the PublishTestResult@2 task with the message "Failed find: ENOENT: no such file or directory, lstat 'C:\agent\_work\2\s\ProjectName.project.~u'"

01 ##[section]Starting: Publish TcUnit results
02 ==============================================================================
03 Task         : Publish Test Results
04 Description  : Publish test results to Azure Pipelines
05 Version      : 2.210.0
06 Author       : Microsoft Corporation
07 Help         : https://docs.microsoft.com/azure/devops/pipelines/tasks/test/publish-test-results
08 ==============================================================================
09 ##[error]Error: Failed find: ENOENT: no such file or directory, lstat 'C:\agent\\_work\2\s\ProjectName.project.~u'
10 ##[section]Finishing: Publish TcUnit results

Did somebody already has the same problem with the publishing of the test results? I can imagine it is more a general thing with running pipelines and publish the test results on Azure. But unfortunately the community for CI workflows with the TwinCAT environment is still quite small. So I hope that someone here can help me how to fix or avoid this error. Maybe it is a timing error?

I am looking forward to great hints that can help us.

**Comment by PTKu** [comment]
Not much knowledge about the implementation here, but from my experience, if there is any use of AutomationInterface, that is a recipe for sporadic instability. Very much as you describe. The thing with `*.~u` file could be that `devenv` process to which the automation interface is attached is closed prematurely for some reason. Just guessing...

**Reply by sagatowski** [reply]
@GitSteller did you manage to solve it?

I think @PTKu is on the right track unforrtunately. Everything going through the automation interface is sporadic.
